src/views/components/survey_components/question_time_setting.vue

```diff
@@ -0,0 +1,368 @@
+    data: function () {
+        return {
+            survey_package: null,
+            num_locked: 0,
+            selected: null,
+            unselected: null,
+            survey_id: null,
+            name: null,
+            edit_click: false,
+            checkbox_indices: [],
+            check_status: [0, 0],
+            term: "2018Winter",
+            start_time: "2000-01-01T00:00:00",
+            end_time: "2000-01-01T00:00:00",
+            // The questions is gonna be a list of question object
+            questions: null,
+            // The backup questions is gonna be a list of back up questions
+            backup_questions: []
+        }
+    },
+    created: async function () {
+        this.survey_id = this.$route.query.survey_id;
+        this.name = this.$route.query.name;
+        // Fetch the list of questions from the API call
+        let fetched = await fetch(
+            "http://localhost:3000/get_info.php?what=surveys&user_id=admin0&survey_id=1"
+        );
+        let fetchedJSON = await fetched.json();
+        console.log(fetchedJSON);
+        // Parse the survey package returned back into the data
+        if (fetchedJSON.TYPE == "error") {
+            // TODO: redirect to the 404 page
+            return 404;
+        } else if (fetchedJSON.TYPE == "survey package") {
+            // assign value to the 'num_locked'
+            this.survey_package = fetchedJSON.DATA;
+            if (fetchedJSON.DATA.dept_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.dept_number_locked);
+            }
+            if (fetchedJSON.DATA.course_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.course_number_locked);
+            }
+            // Assign question package to this.questions
+            this.questions = fetchedJSON.DATA.questions;
+            // Sort the questions in the list to make it obey the order of positions
+            this.questions.sort(this.compare("position"));
+            // Determine the indices that should contain checkboxes
+            this.checkbox_indices.push(this.num_locked);
+            this.checkbox_indices.push(this.num_locked + 1);
+
+            // Set the default survey start and end time according to the JSON data
+            // sent back
+            let date;
+            let time;
+            [date, time] = fetchedJSON.DATA.start_time.split(" ");
+            this.start_time = date + "T" + time;
+            [date, time] = fetchedJSON.DATA.end_time.split(" ");
+            this.end_time = date + "T" + time;
+        }
+        // Fetch all the questions back from the API
+        fetched = await fetch(
+            "http://localhost:3000/get_info.php?what=questions"
+        );
+        fetchedJSON = await fetched.json();
+        console.log(fetchedJSON);
+        // Determine which question objects to put into the backup_questions list
+        for (let i = 0; i < fetchedJSON.length; i++) {
```

Comment:
Lookup the `filter` method on javascript arrays. It'll make this nicer. (hint: You can use `arr.indexOf(...) >= 0` to tell if something is in another array.)

Reply:
I will look up this.